src/test/java/uk/gov/hmcts/reform/roleassignment/domain/service/drools/DroolJudicialCategoryCaseRoleTest.java

```diff
@@ -0,0 +1,461 @@
+                                                                       Status.REJECTED,
+                                                                       roleAssignment.getStatus()
+                                                                   )
+        );
+    }
+
+    @Test
+    void shouldRejectRequestedRoleForNullMandatoryFiled_Jurisdiction() {
+
+        RoleAssignment requestedRole1 =  getRequestedCaseRole_2(RoleCategory.JUDICIAL,
+                                                                "case-allocator",
+                                                                GrantType.SPECIFIC,
+                                                                RoleType.CASE,
+                                                                Classification.PUBLIC,
+                                                                Status.CREATE_REQUESTED);
+        //Jurisdiction is a mandatory field - not provided
+        requestedRole1.getAttributes().put("caseId", convertValueJsonNode("1234567890123456"));
+        requestedRole1.getAttributes().put("caseType", convertValueJsonNode("Asylum"));
+
+        List<RoleAssignment> requestedRoles = new ArrayList<>();
+        requestedRoles.add(requestedRole1);
+
+        assignmentRequest.setRequestedRoles(requestedRoles);
+
+        //Execute Kie session
+        buildExecuteKieSession();
+
+        //assertion
+        assignmentRequest.getRequestedRoles().stream().forEach(roleAssignment ->
+                                                                   assertEquals(
+                                                                       Status.REJECTED,
+                                                                       roleAssignment.getStatus()
+                                                                   )
+        );
+    }
+
+    @Test
+    void shouldRejectRequestedRoleForNullMandatoryField_CaseId() {
+
+        RoleAssignment requestedRole1 =  getRequestedCaseRole(RoleCategory.JUDICIAL, "judge", GrantType.SPECIFIC);
+
+        //CaseId is a mandatory field - not provided
+
+        List<RoleAssignment> requestedRoles = new ArrayList<>();
+        requestedRoles.add(requestedRole1);
+
+        assignmentRequest.setRequestedRoles(requestedRoles);
+        assignmentRequest.getRequestedRoles().stream().forEach(roleAssignment -> {
+
+            roleAssignment.setBeginTime(ZonedDateTime.now(ZoneOffset.UTC));
+        });
+
+        //Execute Kie session
+        buildExecuteKieSession();
+
+        //assertion
+        assignmentRequest.getRequestedRoles().stream().forEach(roleAssignment ->
+                                                                   assertEquals(
+                                                                       Status.REJECTED,
+                                                                       roleAssignment.getStatus()
+                                                                   )
+        );
+    }
```

Comment:
Good to write below scenarios
1. with IAC_1_1 flag off.
2. Invalid assignerId will try to delete and create these roles

Reply:
Have addressed (1.), (2.) is having issues with delete role with wrong assignerId

Reply:
All addressed now